classes/sota_raspberrypi.bbclass

```diff
@@ -21,20 +21,50 @@ UBOOT_ENTRYPOINT_sota ?= "0x00008000"
 IMAGE_FSTYPES_remove_sota = "rpi-sdimg"
 OSTREE_BOOTLOADER ?= "u-boot"
 
+def make_dtb_boot_files(d):
+    # Generate IMAGE_BOOT_FILES entries for device tree files listed in
+    # KERNEL_DEVICETREE.
+    #
+    # This function was taken from conf/machine/include/rpi-base.inc in
+    # meta-raspberrypi
+    alldtbs = d.getVar('KERNEL_DEVICETREE')
+    imgtyp = d.getVar('KERNEL_IMAGETYPE')
+
+    def transform(dtb):
+        base = os.path.basename(dtb)
+        if dtb.endswith('dtb'):
+            return base
+        elif dtb.endswith('dtbo'):
+            return '{};{}'.format(base, dtb)
+
+    return ' '.join([transform(dtb) for dtb in alldtbs.split(' ') if dtb])
+
+IMAGE_BOOT_FILES_sota = "bcm2835-bootfiles/* \
+                         u-boot.bin;${SDIMG_KERNELIMAGE} \
+                         "
```

Comment:
Does this also need to be rpi4-specific as well?

Reply:
No it's needed and was there before. It just has been split in several lines in my version.